src/main/java/com/microsoft/graph/content/MSBatchResponseContent.java

```diff
@@ -47,52 +47,54 @@ public Response getResponseById(final String requestId) {
 		final JsonArray responses = batchResponseArray;
 
 		for (final JsonElement response : responses) {
-			final JsonObject jsonresponse = response.getAsJsonObject();
-			final JsonElement idElement = jsonresponse.get("id");
-			if (idElement != null) {
-				final String id = idElement.getAsString();
-				if (id.compareTo(requestId) == 0) {
-					final Response.Builder builder = new Response.Builder();
-
-					// Put corresponding request into the constructed response
-					builder.request(batchRequestsHashMap.get(requestId));
-					// copy protocol and message same as of batch response
-					builder.protocol(batchResponse.protocol());
-					builder.message(batchResponse.message());
-
-					// Put status code of the corresponding request in JsonArray
-					final JsonElement statusElement = jsonresponse.get("status");
-					if (statusElement != null) {
-						final Long status = statusElement.getAsLong();
-						builder.code(status.intValue());
-					}
+			if(response.isJsonObject()) {
```

Comment:
nit: same early elimination of error condition can be applied with
```java
if (!response.isJsonObject())
    continue;
```

There is a line that starts with 10 tabs in this function, we can keep it at 9, which was already too much :)